src/analysis/function_analysis.cpp

```diff
@@ -376,7 +378,8 @@ DefinednessAnalysis::DefinednessAnalysis(const ParamNames& arg_names, CFG* cfg,
     for (const auto& p : results) {
         RequiredSet required;
         for (const auto& p2 : p.second) {
-            if (scope_info->refersToGlobal(p2.first))
+            if (scope_info->refersToGlobal(p2.first)
+                || scope_info->getScopeTypeOfName(p2.first) == ScopeInfo::VarScopeType::NAME)
```

Comment:
Yeah I can clean this up.

As for whether it should `scope_type != FAST`, well, the `if` statement excluded closure variables before, and it still does. If that's correct, I don't know... I think you would know better than I would.